web/modules/custom/asset_distribution/asset_distribution.behat.inc

```diff
@@ -0,0 +1,302 @@
+   *   The title of the asset distribution.
+   *
+   * @When (I )delete the :asset_distribution asset distribution
+   */
+  public function deleteAssetDistribution($asset_distribution) {
+    /** @var \Drupal\rdf_entity\Entity\Rdf $asset_distribution */
+    $this->getAssetDistributionByName($asset_distribution)->delete();
+  }
+
+  /**
+   * Returns the asset distribution with the given title.
+   *
+   * If multiple asset distributions have the same title,
+   * the first one will be returned.
+   *
+   * @param string $title
+   *   The asset distribution title.
+   *
+   * @return \Drupal\rdf_entity\Entity\Rdf
+   *   The asset distribution.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when an asset distribution with the given title does not exist.
+   */
+  protected function getAssetDistributionByName($title) {
+    return $this->getRdfEntityByLabel($title, 'asset_distribution');
+  }
+
+  /**
+   * Returns the Solution with the given title.
+   *
+   * If multiple solution have the same title, the first one will be returned.
+   *
+   * @param string $title
+   *   The solution title.
+   *
+   * @return \Drupal\rdf_entity\Entity\Rdf
+   *   The solution.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when a solution with the given title does not exist.
+   */
+  public static function getSolutionByName($title) {
+    return self::getRdfEntityByLabel($title, 'solution');
+  }
+
+  /**
+   * Checks the number of available asset distributions.
+   *
+   * @param int $number
+   *   The expected number of asset distributions.
+   *
+   * @Then I should have :number asset distribution(s)
+   */
+  public function assertAssetDistributionCount($number) {
+    $this->assertRdfEntityCount($number, 'asset_distribution');
+  }
+
+  /**
+   * Checks if the given asset distributions belongs to the given solution.
+   *
+   * If there are multiple solutions or asset distributions with
+   * the same title, then only the first one is checked.
+   *
+   * @param string $asset_distribution
+   *   The title of the asset distribution to check.
+   * @param string $solution
+   *   The title of the custom page to check.
+   *
+   * @throws \Exception
+   *   Thrown when an asset distribution with the given title doesn't exist.
+   *
+   * @Then the :asset_distribution asset distribution is related to the :solution solution
+   */
+  public function assertAssetRelationship($asset_distribution, $solution) {
+    /** @var \Drupal\rdf_entity\RdfInterface $asset_distribution */
+    $asset_distribution = $this->getAssetDistributionByName($asset_distribution);
+
+    /** @var \Drupal\rdf_entity\RdfInterface $solution */
+    $solution = $this->getRdfEntityByLabelUnchanged($solution, 'solution');
+
+    // This handles the solutions added by form submit.
+    if (!in_array($asset_distribution, $this->assetDistributions)) {
+      $this->assetDistributions[$asset_distribution->id()] = $asset_distribution;
+    }
```

Comment:
This should not be here. This is an assert (starts with the keyword `@Then`), so this should only check if any visible results are as expected. It should not do any unrelated logic. If a test creates an asset distribution by submitting the form through the interface, they should also remove it again through the interface.